apps/sanity-studio/package.json

```diff
@@ -28,8 +28,8 @@
   },
   "devDependencies": {
     "@sanity/eslint-config-studio": "^4.0.0",
-    "@types/react": "^18.0.25",
+    "@types/react": "^18.3.3",
     "eslint": "^8.6.0",
-    "typescript": "^5.1.6"
+    "typescript": "5.5.4"
```

Comment:
Why pinning?

Reply:
TypeScript doesn't follow semantic versioning and there are often breaking changes on minor version bumps: https://github.com/microsoft/TypeScript/issues/14116

Reply:
Is it a problem? The renovate bot PR will just fail in that case and then we can fix the type errors in that PR before it's merged.

Reply:
Yes, we've had problems in the past where `@sanity/pkg-utils` would type check differently than `tsc --noEmit` due to different TS versions sneaking into the codebase that are all within the same major, but on different minors. Being explicit ensures that if anyone in the future were to nuke the `pnpm-lock.yaml` file and regen it, then you won't suddenly have build errors that you didn't have before, just because a new minor of TypeScript is pulled in that haven't been tested yet.
The philosophy is that doing things like nuking the lockfile at any time on `main` should be safe to do and not introduce unexpected behavior. And AFAIK that requires `typescript` to be pinned 😅

Reply:
That makes sense. But does that then mean that we need to bump `typescript` manually from now on?

Reply:
Renovate handles it ☺️

Reply:
OK, I got no problem then 😆